AppCenter/AppCenter/Internals/Context/AuthToken/MSAuthTokenContext.m

```diff
@@ -222,7 +222,7 @@ - (void)setAuthTokenHistory:(nullable NSArray<MSAuthTokenInfo *> *)authTokenHist
     [MS_USER_DEFAULTS setObject:encryptedData forKey:kMSAuthTokenHistoryKey];
     MSLogDebug([MSAppCenter logTag], @"Saved new history state.");
   } else {
-    MSLogWarning([MSAppCenter logTag], @"Failed to save new history state.");
+    MSLogInfo([MSAppCenter logTag], @"Failed to save new history state.");
```

Comment:
I don't think we want to change the log level when failed to save token in storage. Warning level is still appropriate.

Reply:
In some customer complains, this is also complained, should we fix that as well?

Reply:
okay, that's fair, then we need to address this code path as well.

Reply:
I don't remember exactly but since authTokenHistory can be null it sounds to me that passing nil or empty array to this method is indeed not a warning and can be expected though if the encryption or persisting a valid value is failing then it's a warning.

Reply:
> I don't remember exactly but since authTokenHistory can be null it sounds to me that passing nil or empty array to this method is indeed not a warning and can be expected though if the encryption or persisting a valid value is failin

Yeah, I do see some caller is setting nil. will seperate that out.